sdk/package/sdk.ts

```diff
@@ -0,0 +1,726 @@
+    } catch (e) {
+        console.log("Failed to fetch public balance.");
+        console.error(e);
+    }
+}
+
+/// Converts a given zkAddress to json.
+function privateAddressToJson(privateAddress: Address): Address {
+    const bytes = base58Decode(privateAddress);
+    return JSON.stringify({
+        // spend: Array.from(bytes.slice(0, 32)),
+        // view: Array.from(bytes.slice(32))
+        receiving_key: Array.from(bytes)
+    });
+};
+
+/// Initial synchronization with signer.
+async function init_sync(wasm: any, wasmWallet: Wallet, network: Network): Promise<void> {
+    console.log('Beginning initial sync');
+    const networkType = wasm.Network.from_string(`"${network}"`);
+    const startTime = performance.now();
+    await wasmWallet.restart(networkType);
+    const endTime = performance.now();
+    console.log(
+        `Initial sync finished in ${(endTime - startTime) / 1000} seconds`
+    );
+}
+
+/// Syncs wallet with ledger.
+async function sync(wasm: any, wasmWallet: Wallet, network: Network): Promise<void> {
+    try {
+        console.log('Beginning sync');
+        const startTime = performance.now();
+        const networkType = wasm.Network.from_string(`"${network}"`);
+        await wasmWallet.sync(networkType);
+        const endTime = performance.now();
+        console.log(`Sync finished in ${(endTime - startTime) / 1000} seconds`);
+      } catch (error) {
+        console.error('Sync failed', error);
+      }
+}
+
+/// Attempts to execute a "To Private" transaction by a post on the currently
+/// connected wallet.
+async function to_private_by_post(wasm: any, wasmWallet: Wallet, asset_id: AssetId, to_private_amount: TransferAmount, network: Network): Promise<any> {
+    console.log("to_private transaction...");
+    // let asset = asset_id.toU8a(); // [u8; 32]
+    // const asset_arr = Array.from(uint8);
+    const amountBN = new BN(to_private_amount);
+    const asset_id_arr = Array.from(asset_id);
+    const txJson = `{ "ToPrivate": { "id": [${asset_id_arr}], "value": ${amountBN} }}`;
+    console.log("txJson:" + txJson);
+    const transaction = wasm.Transaction.from_string(txJson);
+    console.log("transaction:" + JSON.stringify(transaction));
+    const networkType = wasm.Network.from_string(`"${network}"`);
+    try {
+        const res = await wasmWallet.post(transaction, null, networkType);
+        console.log("📜to_private result:" + res);
+        return res;
+    } catch (error) {
+        console.error('Transaction failed', error);
+    }
+}
+
+/// Attempts to execute a "To Private" transaction by a sign + sign_and_send on
+/// the currently connected wallet.
+/// Optional: The `onlySign` flag allows for the ability to sign and return
+/// the transaction without posting it to the ledger.
+async function to_private_by_sign(api: ApiPromise, signer: string, wasm: any, wasmWallet: Wallet, asset_id: AssetId, to_private_amount: TransferAmount, network: Network, onlySign: boolean): Promise<any> {
```

Comment:
Instead of having this `onlySign` flag on these transaction functions, I think it would be better to factor out a function that only builds and signs transactions with signer but doesn't publish them, i.e. .the onlySign=true code path. If a user wants to only sign, they can call that function (could call it `buildToPrivate` or something). Otherwise they can call a function called `sendToPrivate` which internally calls  `buildToPrivate` and then publishes the transaction.